smooth-app-bar-layout/src/main/java/me/henrytao/smoothappbarlayout/SmoothAppBarLayout.java

```diff
@@ -151,6 +151,7 @@ public int getCurrentOffset() {
 
   public void setScrollTargetCallback(ScrollTargetCallback scrollTargetCallback) {
     mScrollTargetCallback = scrollTargetCallback;
+    setBehaviorScrollTargetCallback();
```

Comment:
@henrytao-me I understand the intention of this method, but I just don't understand how this is achieved.

> Am I missing something or in your original implementation it does nothing but storing a reference to the passed callback which is never used?

You didn't answer this question. Once again, maybe there's some hidden magic, that I'm not aware of, but in your current implementation the method [setScrollTargetCallback](https://github.com/henrytao-me/smooth-app-bar-layout/blob/v23.2.1/smooth-app-bar-layout/src/main/java/me/henrytao/smoothappbarlayout/SmoothAppBarLayout.java#L153
) does nothing but storing a callback that is literally never used, so scroll target is always ignored unless you explicitly set it to layout's behavior, which isn't that convenient. That's why I added  `setBehaviorScrollTargetCallback ` method. In `setScrollTargetCallback ` it's called to pass that callback to behavior, which will be null if the layout wan't measured, so there's an additional call to this method in behavior's `onInit ` in case that callback was set for layout before behavior was set up.